spark8t/cli/__init__.py

```diff
@@ -1,5 +1,134 @@
+        default="ERROR",
+        help="Set the log level of the logging",
+    )
+
+    return parser
+
+
+def spark_user_parser(parser: ArgumentParser) -> ArgumentParser:
+    """
+    Add Spark user related argument parsing to the existing parser context
+
+    :param parser: Input parser to decorate with parsing support for Spark params.
+    """
+    parser.add_argument(
+        "--username",
+        default="spark",
+        type=str,
+        help="Service account name to use other than primary.",
+    )
+    parser.add_argument(
+        "--namespace",
+        default="default",
+        type=str,
+        help="Namespace of service account name to use other than primary.",
+    )
+    return parser
+
+
+def k8s_parser(parser: ArgumentParser) -> ArgumentParser:
+    """
+    Add K8s related argument parsing to the existing parser context
+
+    :param parser: Input parser to decorate with parsing support for Spark params.
+    """
+    parser.add_argument(
+        "--master", default=None, type=str, help="Kubernetes control plane uri."
+    )
+    parser.add_argument(
+        "--kubeconfig", default=None, type=str, help="Kubernetes configuration file"
+    )
+    parser.add_argument(
+        "--context", default=None, type=str, help="Kubernetes context to be used"
+    )
+    parser.add_argument(
+        "--backend",
+        default="kubectl",
+        choices=["kubectl", "lightkube"],
+        type=str,
+        help="Kind of backend to be used for talking to K8s",
+    )
+    return parser
+
+
+def add_config_arguments(parser: ArgumentParser) -> ArgumentParser:
+    """
+    Add arguments to provide extra configurations for the spark properties
+
+    :param parser: Input parser to decorate with parsing support for deploy arguments.
+    """
+    parser.add_argument(
+        "--properties-file",
+        default=None,
+        type=str,
+        help="Spark default configuration properties file.",
+    )
+    parser.add_argument(
+        "--conf",
+        action="append",
+        type=str,
+        help="Config properties to be added to the service account.",
+    )
+    return parser
+
+
+def add_deploy_arguments(parser: ArgumentParser) -> ArgumentParser:
+    """
+    Add deployment related argument parsing to the existing parser context
+
+    :param parser: Input parser to decorate with parsing support for deploy arguments.
+    """
+    parser.add_argument(
+        "--deploy-mode",
+        default="cluster",
+        type=str,
+        help="Deployment mode for job submission. Default is 'client'.",
+        choices=["client", "cluster"],
+    )
+    return parser
+
+
+def get_kube_interface(args: Namespace) -> AbstractKubeInterface:
+    return (
+        LightKube(args.kubeconfig or defaults.kube_config, defaults)
+        if args.backend == "lightkube"
+        else KubeInterface(
+            args.kubeconfig or defaults.kube_config, context_name=args.context
+        )
+    )
```

Comment:
I really don't think this should belong to `__init__.py`...

Reply:
I'd almost say 'anywhere else' ;-) 
`utils.py` or `helpers.py` is still better than "secretly hiding" a bunch of utility/helper functions in `__init__.py`.
"Explicit is always better then implicit" (Zen of Python)

What exactly is the problem with `utils.py`? 
If it's the lenght: we may wanna go with my suggestion from earlier, regarding a common interface/parent_class for those of our scripts/classes that have a decent "common denominator". I guess a good number of the current `utils.py` may be good to go on that class.

(Let me take a look a the actual functionlaities declared in this module, I may have more suggestions then the "blnind" one above.)

Reply:
I'll make a separate module on cli named confs. I don't mind too much. These bits really only pertain to `cli`: that's why it is good place there. But storing stuff on `__init__` is not that strange or hidden, really. It is just something that would get imported by default in the submodule, which is not too far from reality here given all scripts need these confs

Reply:
I gave quite a bit of thoughts on this. If possible, I'd avoid using "_". I'm rather happy with `args` TBH, given that the module is about *command-line arguments* functionalities, which is really clear from the path `cli.args`. I also thought about `cli.argparse` but I'd rather NOT have argparse conflicts with the package.

Reply:
We definately should avoid interference with existing Python packages. 
(Which is probably part of the reason why I'm a bit uncomfortable to leave possibility for kinda stuff like below
```
from cli import args

def f(*args, **kwargs):
    args.parse(args)
)
```
Is there any particular reason why we may want to avoid underscores in module names? I see perfectly valid reason for package names, but not for modules. 

"Modules should have short, all-lowercase names. Underscores can be used in the module name if it improves readability. Python packages should also have short, all-lowercase names, although the use of underscores is discouraged." ([PEP8: Package and Module Names](https://peps.python.org/pep-0008/#package-and-module-names))

Reply:
I think `args` is not a good name. I will suggest to have something like: `arguments` or `parser_arguments` or something similar. What do you think @juditnovak @deusebio ?

Reply:
I think we could choose from 2 strategies here.

1. A lightweight module strictly dealing with command line parameters to be added to our `argparse` parser. In this case `cli.parameters` or `cli.parse_params` could be a descriptive name. 
  (*Note* : I guess we should rather use 'parameter' instead of 'argument', no? https://stackoverflow.com/questions/36495669/difference-between-terms-option-argument-and-parameter)
2. We set up a module to deal with all our parameter parsing logic, if more than "`parser.add()`". Typically if we have code for config file parsing, defaults parsed from ENV_VARs, priority across potentially multiple definitions, etc. I mean the whole parameter parsing feature we've implemented for the `spark8t` tool.
In this case something referring to this feature (facilitating the usage of complicated Spark parameters) may be better? Like `spark8t.cli.easy_params`, or `priority_params` (suggesting that they can be defined on mulitple levels taken by strict precedence), may refer better to the logical unit (i.e. feature implementation)...?

Reply:
Shorty, responding to @welpaolo :-)  If we choose to go with (1), then I think we're totally on the same page -- I'm just wondering that `params` may be better to use instead? (As we're dealing with parameters not arguments.) So rather `parse_params`...?)

Reply:
I feel we are a bit overthinking this. Let's go with `cli.params` which seems to fit a bit all preferences/suggestions/concerns